plugins/modules/primary_ip.py

```diff
@@ -163,13 +212,22 @@ def _get_primary_ip(self):
             self.fail_json_hcloud(exception)
 
     def _create_primary_ip(self):
-        self.module.fail_on_missing_params(required_params=["type", "datacenter"])
+        self.module.fail_on_missing_params(required_params=["type", "name"])
+        try:
+            check_required_one_of([["server", "datacenter"]], {k: v for k, v in self.module.params.items() if v})
+        except TypeError as e:
+            self.module.fail_json(msg=to_native(e))
```

Comment:
I opened https://github.com/ansible/ansible/issues/82724 to see if we can improve this.

Reply:
Let's implement our own and move this code to a helper method in the parent class, as I doubt ansible-core will ship the fix in the next weeks. @jjeff07 still up to continue this PR?